notification_configuration.go

```diff
@@ -93,12 +92,27 @@ type NotificationConfiguration struct {
 
 // DeliveryResponse represents a notification configuration delivery response.
 type DeliveryResponse struct {
-	Body       string      `json:"body"`
-	Code       int         `json:"code"`
-	Headers    http.Header `json:"headers"`
-	SentAt     time.Time   `json:"sent-at,iso8601"`
-	Successful bool        `json:"successful"`
-	URL        string      `json:"url"`
+	Body       string    `jsonapi:"attr,body"`
+	Code       string    `jsonapi:"attr,code"`
+	Headers    headers   `jsonapi:"attr,headers"`
```

Comment:
Changed to use the `headers` struct below instead of `http.Header` because jsonapi does not, currently, handle `map[string][]string`. Also, the headers listed in the struct below match what is listed in the documentation. This is more orderly than an arbitrary list of headers. If a customer wants to add another header, that can very easily be accommodated.

Reply:
(I also wondered how the tests pass with this, given that it is a bigger behavior change; [turns out we don't actually test beyond equality](https://github.com/hashicorp/go-tfe/blob/074a0e44b072d8f9174639b0139cf1e9fb2f8e2e/notification_configuration_test.go#L167-L168) 😕)

Reply:
I don't know how this actually ever worked. Even in the outdated fork, along with upstream, if the tag for the struct isn't `jsonapi`, [it gets skipped](https://github.com/svanharmelen/jsonapi/blob/master/request.go#L156-L159).

Reply:
Once [this PR](https://github.com/hashicorp/jsonapi/pull/2) is merged, i'll update this and we'd be good to go!

Reply:
This if fixed!!